src/include/platform/internal/GenericConfigurationManagerImpl.ipp

```diff
@@ -941,15 +941,12 @@ CHIP_ERROR GenericConfigurationManagerImpl<ImplClass>::_ComputeProvisioningHash(
         // (This will also be used for the private key).
         if (certsLen > dataBufSize)
         {
-            chip::Platform::MemoryFree(dataBuf);
-
             dataBufSize = certsLen;
-            dataBuf     = (uint8_t *) chip::Platform::MemoryAlloc(dataBufSize);
-            VerifyOrExit(dataBuf != NULL, err = CHIP_ERROR_NO_MEMORY);
+            VerifyOrExit(dataBuf.Alloc(dataBufSize), err = CHIP_ERROR_NO_MEMORY);
         }
 
         // Read the device intermediate CA certificates.
-        err = Impl()->_GetManufacturerDeviceIntermediateCACerts(dataBuf, certsLen, certsLen);
+        err = Impl()->_GetManufacturerDeviceIntermediateCACerts(dataBuf.Ptr<uint8_t>(), certsLen, certsLen);
```

Comment:
why is this <uint8_t> templating necessary here, but not below for ClearSecretData?

Reply:
I tried verbatim copy. Data was uint8_t by default hence conversions everywhere, but clear secret data takes void* I think. I did not check the rest.

Speaking of which: clearsecretdata is called on the last free, however we have an internal alloc that does NOT clear it. Unsure if intentional or not, so I kept 'as is' but I do assume a bug where using this class would have made easier to spot.

Reply:
clearsecretdata() takes a uint8_t*

Reply:
Hmmm... new theory (verified though): when I do 'run functional and unit tests' in vscode, this file does not get called at all, so it does not notice that the clearsecretdata call is bad.

I updated it blindly and will rely on CI to check things. This is very odd... is this file even used?